packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -4221,6 +4249,16 @@ namespace Tpetra {
           rowValues[j] *= static_cast<impl_scalar_type> (vectorVals[colInds[j]]);
         }
       }
+
+      // FENCE REVIEW - NOT TESTED
+      //   Testing: This code is NOT exercised by unit tests.
+      //   GTX960:  N/A
+      //   White:   N/A
+      //   Plan:    Potentially add unit testing for this code.
+      //            Potentially remove this fence but prior code appears
+      //            to need refactoring.
+      //   Notes:
+
```

Comment:
Same as above. We need this fence or the one in the above if we can establish a design rule.

Reply:
@csiefer2 @jhux2 @kyungjoo-kim 
I agree with @kyungjoo's request for a design rule, and with the decision we made last week.  In the case of CrsMatrix leftScale and rightScale, however, I question whether we actually need the branch that contains the fences.

This branch is executed if the CrsMatrix does not have a localMatrix yet.  A CrsMatrix has a localMatrix if it has ever been fillCompleted OR if it was constructed using a localMatrix.  Thus, this branch gets executed only for a non-fillComplete matrix constructed with something other than a localMatrix (e.g., with insertGlobalValues).  As @MicheldeMessieres ' experiments revealed, we do not test the case of leftScaling or rightScaling a never-fillCompleted matrix.  

Rather than remove one fence and add tests to cover this branch, I wonder if we should support leftScale and rightScale only for fillComplete matrices.   Are there use-cases in Ifpack2, MueLu or Tpetra's MatrixMatrix routines that rely on scaling non-fillComplete matrices?  

I appreciate your thoughts.  In the meantime, I'll fire up a test with leftScale and rightScale functional only for fillComplete matrices, and see if anything downstream breaks.  Thanks!

Reply:
Removed with PR 7560